installinator-artifactd/src/http_entrypoints.rs

```diff
@@ -32,10 +33,13 @@ pub fn api() -> ArtifactServerApiDesc {
 /// Fetch an artifact from the in-memory cache.
 #[endpoint {
     method = GET,
-    path = "/artifacts/{name}/{version}"
+    path = "/artifacts/{kind}/{name}/{version}"
 }]
 async fn get_artifact(
     rqctx: RequestContext<ServerContext>,
+    // NOTE: this is an `ArtifactId` and not an `UpdateArtifactId`, because this
+    // code might be dealing with an unknown artifact kind. This can happen
+    // if a new artifact kind is introduced across version changes.
     path: Path<ArtifactId>,
```

Comment:
Q for @iliana and @david-crespo: we introduce an `ArtifactId` here that is similar to `UpdateArtifactId`, except its kind is potentially unknown (unlike `KnownArtifactKind` which is an enum, `ArtifactKind` is just a newtype wrapper around a string).

For wicketd it makes sense to use that because we might be introducing new artifact kinds across versions, and wicketd is mostly a simple shoveler of data rather than something that cares about the content of each artifact.

Do the same considerations apply to Nexus and sled-agent? In other words, should Nexus and sled-agent use this `ArtifactId` rather than `UpdateArtifactId`? We'll still want to have `KnownArtifactKind` be around, but it wouldn't be a type that gets transmitted over the wire. As a bonus, it would also eliminate a big chunk of the complexity involved in adding a new known artifact kind.